llama/model.py

```diff
@@ -177,7 +177,7 @@ def forward(
         scores = torch.matmul(xq, keys.transpose(2, 3)) / math.sqrt(self.head_dim)
         if mask is not None:
             scores = scores + mask  # (bs, n_local_heads, seqlen, cache_len + seqlen)
-        scores = F.softmax(scores.float(), dim=-1).type_as(xq)
+        scores = F.softmax(scores.type_as(xq), dim=-1).type_as(xq)
```

Comment:
as `xq` and `scores` are both `float16`, and as `F.softmax` returns a `float16` when inputted a `float16`, why not simplifying by

```
scores = F.softmax(scores, dim=-1)
```